### ISSUE TITLE
devcontainer exec attempts confuses args depending on order

### ISSUE THREAD

**Issue by csweichel** [devcontainers/cli#657]
When running `devcontainer exec --workspace-folder . ls` everything works as expected.
When running `devcontainer --workspace-folder . exec ls` the devcontainer CLI attempts to run `exec` in the container. It confuses the args and interprets `exec` as `cmd`.

![image](https://github.com/devcontainers/cli/assets/3210701/8b655e87-52a0-4bf2-b6ac-d802835b1216)

Am I correct in assuming that `devcontainer --workspace-folder . exec` should be a valid way to invoke exec?

**Comment by samruddhikhandale** [comment]
Hi 👋 

> Am I correct in assuming that devcontainer --workspace-folder . exec should be a valid way to invoke exec?

I don't think that the `@devcontainers/cli` is configured to be used in ^ way, and that's the reason why it's currently failing.

The `@devcontainers/cli` expects it to be used as `devcontainer <command> [optional args]`. See 👇 

![image](https://github.com/devcontainers/cli/assets/24955913/85784626-8a62-498e-b79f-e4e40e53caa9)

> Am I correct in assuming that devcontainer --workspace-folder . exec should be a valid way to invoke exec?

Curious, @csweichel is there a specific reason/interest in expecting it ^ to work in such a way? We can definitely update the CLI based on community interest/feedback, but doesn't seem like this is planned for now.

**Comment by csweichel** [comment]
> Curious, @csweichel is there a specific reason/interest in expecting it ^ to work in such a way? We can definitely update the CLI based on community interest/feedback, but doesn't seem like this is planned for now.

Largely past experience with other CLIs where more often than not the order of flags and args does not matter all that much, the occasional ordering of global/command-specific flags notwithstanding.

That said, I'd have expected the CLI to fail with a definite error message when used "incorrectly", rather than it trying to execute something. I was surprised because it seemed like it was working, just not correctly.

**Comment by groner** [comment]
`--workspace-folder` is needed for most of the subcommands here, and I have been tempted to use a shell alias or wrapper function to provide it automatically. The most obvious way to do that is 

```bash
alias devcontainer='\devcontainer --workspace-folder=$PROJECT_DIR'
```

In any event, I think this should generate a clear error and not for devcontainer to try to execute a command named `exec` in the dev container.

```bash
:; devcontainer --workspace-folder=. exec id
OCI runtime exec failed: exec failed: unable to start container process: exec: "exec": executable file not found in $PATH: unknown
```

**Comment by moltar** [comment]
Agreed on this too. Found this to be quite confusing and spent a lot of time debugging.

This works:

```sh
devcontainer --workspace-folder=. up
```

But this doesn't:

```sh
devcontainer --workspace-folder=. exec ...
```

I intended to alias or script the CLI with project-specific configuration options.

It would be great to have this support universally, or be able to set these via env vars.
